### PR TITLE
Update permissions for API routes

### PR DESCRIPTION
**Pull request by GregSutcliffe**
Necessary to fix the core routes tests, eg http://ci.theforeman.org/job/test_plugin_foreman_discovery/47/

@lzap how does it look to you?

### CODE REVIEW

File: lib/foreman_discovery/engine.rb [hunk 1]
```diff
@@ -30,14 +30,28 @@ class Engine < ::Rails::Engine
...
-          permission :provision_discovered_hosts, {:discovered_hosts => [:new, :create] }, :resource_type => 'Host'
-          permission :edit_discovered_hosts, {:discovered_hosts => [:edit, :update, :update_multiple_location,
-                                                             :select_multiple_organization,
-                                                             :update_multiple_organization,
-                                                             :select_multiple_location] }, :resource_type => 'Host'
-          permission :destroy_discovered_hosts, {:discovered_hosts => [:destroy, :submit_multiple_destroy, :multiple_destroy] }, :resource_type => 'Host'
-
+          permission :view_discovered_hosts, {
+              :discovered_hosts          => [:index, :show, :auto_complete_search],
+              :"api/v2/discovered_hosts" => [:index, :show, :auto_complete_search]
```
Comment (domcleal) [inline lib/foreman_discovery/engine.rb]: no auto-complete on the API

File: lib/foreman_discovery/engine.rb [hunk 2]
```diff
@@ -30,14 +30,28 @@ class Engine < ::Rails::Engine
...
+          permission :provision_discovered_hosts, {
+              :discovered_hosts          => [:new, :create],
+              :"api/v2/discovered_hosts" => [:create]
+            }, :resource_type => 'Host'
+          permission :edit_discovered_hosts, {
+              :discovered_hosts          => [:edit, :update, :update_multiple_location,
+                                             :select_multiple_organization,
+                                             :update_multiple_organization,
+                                             :select_multiple_location, :refresh_facts],
+              :"api/v2/discovered_hosts" => [:update, :update_multiple_location,
```
Comment (domcleal) [inline lib/foreman_discovery/engine.rb]: no `*multiple*` actions

File: lib/foreman_discovery/engine.rb [hunk 3]
```diff
@@ -30,14 +30,28 @@ class Engine < ::Rails::Engine
...
+                                             :select_multiple_organization,
+                                             :update_multiple_organization,
+                                             :select_multiple_location, :facts]
+            }, :resource_type => 'Host'
+          permission :destroy_discovered_hosts, {
+              :discovered_hosts          => [:destroy, :submit_multiple_destroy, :multiple_destroy],
+              :"api/v2/discovered_hosts" => [:destroy, :submit_multiple_destroy, :multiple_destroy]
```
Comment (domcleal) [inline lib/foreman_discovery/engine.rb]: ditto

**Comment by GregSutcliffe** [comment]
thanks @domcleal, cutnpaste fail. Updated.

**Comment by domcleal** [comment]
Merged as 5c524bb, thanks!

**Comment by lzap** [comment]
So since :new is outside of the scope, image is fine I guess. Sorry for late reply, on the backlog.

**Pull request closed without merging**